docs/DeveloperGuide.md

```diff
@@ -4,31 +4,95 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+The app was mainly inspired by [addressbook-level2](https://github.com/se-edu/addressbook-level2).
+
+### Design
+The App consists of a few main components.
+
+* Ui: The UI that deals with getting user input and displaying output.
+* Command: Command to be executed.
+* Parser: Parse user input and make sense of it.
+* Storage: Reads data from, and writes data to, the hard disk.
+* Rex: The main class that initialize the app and starts running.
+
+The rest are just objects that the app requires.
+
+The diagram below shows the basic design.
+![image info](./pictures/UML.png)
+
+The complete class diagram is shown below.
+![image info](./pictures/classDiagram.PNG)
+
+### Implementation
+#### Retrieve Command
+The flow for the retrieve command is as follows:
+![image info](./pictures/retrieveCommandSD.png)
+1. Retrieve command extracts NRIC using parser, 
+2. Retrieve command searches through the patients list and obtains the patient's index.
+3. Patients list returns the patient.
+4. Ui object prints the patient.
+
+#### Add Command
+The flow for add command is as follows:
+![image_info](./pictures/addCommandSD.PNG)
+1. Add command extracts NRIC by calling its own method.
+2. Add command checks patients list if NRIC already exists. Add command ends if so.
+3. Add command calls Ui object to get the patient name and date of birth from user.
+4. Patient list adds the new patient.
+5. Add command obtains the newly added patient.
+6. Ui receives the call to print the patient.
+7. The updated patients list is saved by the Storage object.
+
+
+#### Book Command
+The flow for the book command is shown in the sequence diagram below:
+![image info](./pictures/bookApptSequenceDiagram.PNG)
```

Comment:
the size of the image is a little bit small and hard to read maybe can try to ajust the size. Also, there is lots of information in the diagram without text explanation, maybe it is better to provide some necessary explanation about your diagram.